stdlib/REPL/test/replcompletions.jl

```diff
@@ -1754,3 +1755,50 @@ end
         @test c == Any["var\"(\""]
     end
 end
+
+# https://github.com/JuliaLang/julia/issues/36437
+struct Issue36437{T}
+    v::T
+end
+Base.propertynames(::Issue36437) = (:a, :b, :c)
+function Base.getproperty(v::Issue36437, s::Symbol)
+    if s === :a
+        return 1
+    elseif s === :b
+        return 2
+    elseif s === :c
+        return getfield(v, :v)
+    else
+        throw(ArgumentError(lazy"`(v::Issue36437).$s` is not supported"))
+    end
+end
+
+struct Issue36437_2{T}
+    v::T
+end
```

Comment:
Is this supposed to be used for anything?

Reply:
Yeah, thanks for point this out!